_pr-doc-deployment/action.yml

```diff
@@ -199,36 +205,72 @@ runs:
           echo "PR documentation folder created."
         fi
 
-    - name: "Ensure existence of the clean sub-directory corresponding to the current PR"
-      id: create-pr-directory
+    - name: "Ensure a clean sub-directory corresponding to the current PR and count number of current deployments"
+      id: create-pull-directory
       shell: bash
       env:
         PR_NUMBER: ${{ github.event.pull_request.number }}
       run: |
         if [[ -d pull/${PR_NUMBER} ]]; then
+          # This means this is a redeployment
+          echo "FIRST_TIME_DEPLOYMENT=false" >> ${GITHUB_OUTPUT}
+
           echo "Directory corresponding to this PR's documentation detected, cleaning ..."
-          rm -rf pull/${PR_NUMBER} && mkdir pull/${PR_NUMBER}
-        else
+          rm -rf pull/${PR_NUMBER}
+
+          # Count the number of deployments before creating the sub-directory
+          # This ensures correct behaviour in the case of redeployments
+          echo "DEPLOYMENT_COUNT=$(ls pull/ | wc -l)" >> ${GITHUB_OUTPUT}
+
           mkdir pull/${PR_NUMBER}
-          echo "Directory for deploying documentation of the PR created."
+        else
           # This means this is the first time the PR documentation is being deployed
           echo "FIRST_TIME_DEPLOYMENT=true" >> ${GITHUB_OUTPUT}
+
+          mkdir pull/${PR_NUMBER}
+          echo "Directory for deploying documentation of the PR created."
+
+          # Count the number of deployments after creating the sub-directory
+          # This ensures accurate behaviour for first time deployments
+          echo "DEPLOYMENT_COUNT=$(ls pull/ | wc -l)" >> ${GITHUB_OUTPUT}
         fi
 
+    - name: "Detect if deployment rejection comment had been previously added to the same PR"
+      if: ${{ fromJSON(steps.create-pull-directory.outputs.DEPLOYMENT_COUNT) > fromJSON(inputs.maximum-pr-doc-deployments) }}
+      id: find-comment
+      uses: peter-evans/find-comment@3eae4d37986fb5a8592848f6a574fdf654e61f9e # v3.1.0
+      with:
+        issue-number: ${{ github.event.pull_request.number }}
+        body-includes: The documentation for this pull request will not be deployed.
```

Comment:
Let's document why it can not be deployed by adding some additional information.

> Preview limit reached: skipping documentation deployment for this pull request.